WordPress/gutenberg
# [RNMobile] Add scrollable screens to the bottom-sheet

GB-mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2741

Motivation:
Since we have navigation inside the bottom-sheet, there is a possibility to have a flat list on some of the screens. In that scenario, we need to remove the scroll view from the bottom-sheet dynamically while switching the screen which generates some issues.

In this PR I moved the ScrollView to the navigation-screen and we can set `isScrollable` prop for each screen. I left the ScrollView only for bottom-sheets w/o navigation inside.

I added the `isScrollable` prop to the `BottomSheet.NavigationScreen` which determines if the content of the screen has FlatList/List/ScrollView inside.

## How has this been tested?
- Open Button Settings and check if the scroll works as before
- Open Menu Inserter and check if the scroll works as before
- Open Alignment or Header type bottom-sheet and check if the scroll works as before
- Open link settings bottom-sheet and check if it works as before

This PR shouldn't bring any changes in the way the editor works - it is only a different implementation. 

## Checklist:
